### PR TITLE
Initial Dapr integration 

### LINKED ISSUE #16: Feature: Add Dapr for unified APIs to connect to infrastructure and promote best cloud native practices
### Have you searched for this feature request?

- [X] I searched but did not find similar requests

### Problem Statement

idpbuilder right now setups tools that help teams to get started with creating cloud resources but it doesn't help development team to connect to these resources. Adding tools like Dapr will help platform teams to enable developers to build complex application that can connect to infrastructure components by using APIs. 

### Possible Solution

Deploy and configure Dapr as part of the stack

### Alternatives Considered

There are not many other projects providing the features that Dapr provides.

### PR DESCRIPTION
**Pull request by salaboy**
Fixes #16 
Hi community, I am just building a simple Dapr integration for CNOE Stacks. 
I would love to build deeper examples with Backstage, but I didn't found out much more resources on that. 
Can you please help me to review this first?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,356 @@
-# CNOE Stacks
+[![Codespell][codespell-badge]][codespell-link]
```
[the remark this answers is not in the document]
Comment (nabuskey) [reply README.md:1]: Agreed. I am open to updating readme in a separate PR/issue but for initial implementation of a particular stack, I'd rather not make major updates to it.

File: dapr-integration/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Dapr Integrations 
+
+`idpBuilder` is extensible to launch custom Dapr patterns using package extensions. 
+
+Please use the below command to deploy an IDP reference implementation with an Argo application for preparing up the setup for terraform integrations:
+
+```bash
+idpbuilder create \
+  --use-path-routing \
+  --package-dir https://github.com/cnoe-io/stacks//ref-implementation \
```
Comment (nabuskey) [inline dapr-integration/README.md:10]: Please use the `-p` flag instead. The `--package-dir` flag is deprecated.

**Review by nabuskey — CHANGES_REQUESTED** [review]

File: dapr-integration/README.md (hunk 2 above)
Comment (salaboy) [reply dapr-integration/README.md]: done

[the remark this answers is not in the document]
Comment (salaboy) [reply dapr-integration/README.md]: done

File: dapr-integration/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  --package-dir https://github.com/cnoe-io/stacks//dapr-integrations
+```
+## What is installed?
+
+1. Dapr Control Plane
+1. Dapr Statestore and PubSub components
+
+This needs your credentials for this to work. Follow the Dapr installation documentation on how to add your credentials.
+
+## Application with cloud resources.
```
[the remark this answers is not in the document]
Comment (salaboy) [reply dapr-integration/README.md]: done

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (salaboy) [reply README.md:1]: so .. a separate PR for updating the README to include this integration

**Comment by salaboy** [comment]
@salaboy @nabuskey I think I made all the requested changes.. I will work on a separate repo for some Dapr Backstage templates.. let me know if you want me to add something else to this PR, I would prefer to keep it simple

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nabuskey) [reply README.md:1]: You just need to update the section that's pointed by Carlos above. I don't think we want to update the main readme file like you have it currently in your branch.

**Comment by salaboy** [comment]
@nabuskey sorry about that.. now it is fixed

File: dapr-integration/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Dapr Integrations 
+
+`idpBuilder` is extensible to launch custom Dapr patterns using package extensions. 
+
+Please use the below command to deploy an IDP reference implementation with an Argo application for preparing up the setup for terraform integrations:
```
Comment (nabuskey) [inline dapr-integration/README.md:5]: Copy paste error? `terraform integrations`

File: dapr-integration/README.md
Comment (nabuskey) [inline dapr-integration/README.md:1]: This readme is light. What do you expect people to do once you have it installed?

**Review by nabuskey — CHANGES_REQUESTED** [review]

File: dapr-integration/README.md (hunk 4 above)
Comment (salaboy) [reply dapr-integration/README.md]: fixed

File: dapr-integration/README.md
Comment (salaboy) [reply dapr-integration/README.md:1]: @nabuskey added more information and links

Comment (nabuskey) [reply dapr-integration/README.md:1]: Can you fix DCO? I think it's good to go after DCO is fixed.

Comment (salaboy) [reply dapr-integration/README.md:1]: done

**Review by nabuskey — APPROVED** [review]

**Comment by nimakaviani** [comment]
Thank you @salaboy this is very cool. Would be good to publish some Backstage templates to go with it for some sample usecase applications similar ot what we have [here](http://localhost:3000/docs/reference-implementation/integrations/terraform/s3-bucket) and I think you and @csantanapr should have a "How to use Dapr in Dev Platforms" session maybe?

**Comment by salaboy** [comment]
I am working on those templates with examples in different languages.
I will keep you posted. Let’s record a video or I can write a blog post
with those examples.



 - Blog: http://salaboy.com <http://salaboy.wordpress.com>
 - Github user: http://github.com/salaboy
 - Twitter: http://twitter.com/salaboy

 - Mauricio "Salaboy" Salatino -

**Comment by nimakaviani** [comment]
awesome sounds good!

**Comment by csantanapr** [comment]
awesome @salaboy

**Pull request merged**